drivers/clock_control/Kconfig.stm32

```
@@ -121,12 +134,14 @@ config CLOCK_STM32_MCO1_SRC_HSI48
 	           SOC_SERIES_STM32H7X || \
 	           SOC_SERIES_STM32H7RSX || \
 	           SOC_SERIES_STM32H5X
+	depends on SOC_SERIES_STM32L4X || SOC_SERIES_STM32U5X
```

Using `depends on` here is erroneous per https://www.kernel.org/doc/html/next/kbuild/kconfig-language.html:
_"If multiple dependencies are defined, they are connected with ‘&&’"_, which is not what we want here.

```suggestion
	           SOC_SERIES_STM32H5X || \
	           SOC_SERIES_STM32L4X || \
	           SOC_SERIES_STM32U5X
```

---

Fixed.